src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/TeasyElementFinder.java

```diff
@@ -0,0 +1,108 @@
+package com.wiley.autotest.selenium.elements.upgrade.v3;
+
+import com.wiley.autotest.selenium.context.SearchStrategy;
+import com.wiley.autotest.selenium.elements.upgrade.TeasyElement;
+import com.wiley.autotest.selenium.elements.upgrade.TeasyElementType;
+import com.wiley.autotest.selenium.elements.upgrade.v3.conditions.FramesConditionFactory;
+import org.openqa.selenium.Alert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.wiley.autotest.selenium.elements.upgrade.TeasyElementType.DOM;
+import static com.wiley.autotest.selenium.elements.upgrade.TeasyElementType.VISIBLE;
+import static com.wiley.autotest.selenium.elements.upgrade.TeasyElementWrapper.wrap;
+import static com.wiley.autotest.selenium.elements.upgrade.TeasyElementWrapper.wrapList;
+
+
+/**
+ * Finds different kinds of elements for example
+ * visible, present in dom, alerts etc.
+ */
+public class TeasyElementFinder {
+
+    private TeasyElement context;
+    private TeasyFluentWait<WebDriver> fluentWait;
+    private FramesConditionFactory conditionFactory;
+
+    public TeasyElementFinder(WebDriver driver, SearchStrategy strategy) {
+        this.fluentWait = new TeasyFluentWait<>(driver, strategy);
+        this.conditionFactory = new FramesConditionFactory(context, strategy.getFrameStrategy());
+    }
+
+    public TeasyElementFinder(WebDriver driver, SearchStrategy strategy, TeasyElement context) {
+        this(driver, strategy);
+        this.context = context;
+    }
+
+    public TeasyElement visibleElement(By locator) {
+        return getElement(locator, conditionFactory.get().visibility(locator), VISIBLE);
+    }
+
+    /**
+     * note: Will return empty list in case no visible elements found
+     */
+    public List<TeasyElement> visibleElements(By locator) {
+        return getElements(locator, conditionFactory.get().visibility(locator), VISIBLE);
+    }
+
+    public TeasyElement presentInDomElement(By locator) {
+        return getElement(locator, conditionFactory.get().presence(locator), DOM);
+    }
+
+    /**
+     * note: Will return empty list in case no visible elements found
```

Comment:
is it the correct description?

Reply:
it s not a full description...more of a note. If you mean that it does not follow java conventions - I agree. but as of now, I am not ready to provide comments to all methods in a standard style. It could be another task -  feel free to submit a pull rquest with it ;)